app/page.jsx

```diff
@@ -17,37 +17,58 @@ const getTickets = async () => {
     return res.json();
   } catch (error) {
     console.log("Error loading tickets: ", error);
-    return { tickets: [] };
+    return { tickets: [] }; 
   }
 };
 
 const Dashboard = () => {
   const [tickets, setTickets] = useState([]);
+  const [filteredTickets, setFilteredTickets] = useState([]);
+  const [statusFilter, setStatusFilter] = useState("all");
   const [viewBy, setViewBy] = useState("category");
 
   useEffect(() => {
     const fetchTickets = async () => {
       const data = await getTickets();
       setTickets(data?.tickets || []);
+      setFilteredTickets(data?.tickets || []);
     };
 
     fetchTickets();
   }, []);
 
-  const uniqueCategories = [...new Set(tickets?.map(({ category }) => category))];
+  useEffect(() => {
+    if (statusFilter === "all") {
+      setFilteredTickets(tickets);
+    } else {
+      setFilteredTickets(
+        tickets.filter((ticket) =>
+          ticket.status.toLowerCase() === statusFilter.toLowerCase()
+        )
+      );
+    }
+  }, [statusFilter, tickets]);
+
+  const uniqueCategories = [...new Set(filteredTickets?.map(({ category }) => category))];
   const uniqueStatuses = [...new Set(tickets?.map(({ status }) => status))];
 
   const handleViewByChange = (event) => {
     setViewBy(event.target.value);
   };
 
+  const handleStatusChange = (event) => {
```

Comment:
please rename this object to handleStatusFilterChange

Reply:
Done